engine/src/main/java/org/destinationsol/assets/audio/OggSoundManager.java

```diff
@@ -163,11 +163,13 @@ public void play(SolGame game, PlayableSound playableSound, @Nullable Vector2 po
         if (source == null && sound.getLoopTime() > 0) {
             throw new AssertionError("Attempted to loop a sound without a parent object: " + sound.getUrn());
         }
+
+        Vector2 soundPosition = position;
         if (position == null) {
-            position = source.getPosition();
+            soundPosition = source.getPosition();
```

Comment:
A good way to ensure that this kind of modification does not happen to parameters it is good practice to declare them as `final`. 😉